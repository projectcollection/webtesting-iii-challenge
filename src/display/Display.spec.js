// Test away!
import React from 'react';
import renderer from 'react-test-renderer';
import {render, cleanup} from 'react-testing-library';
import 'jest-dom/extend-expect'

import Display from './Display'

afterEach(cleanup)

describe('<Display/>', () => {
    test('should match snapshot', () => {
        const tree = renderer.create(<Display/>).toJSON()

        expect(tree).toMatchSnapshot()
    });

    test('should match snapshot', () => {
        const tree = renderer.create(<Display
            locked = {false}
            closed = {false}
            />).toJSON()

        expect(tree).toMatchSnapshot()
    });

    test('should match snapshot', () => {
        const tree = renderer.create(<Display
            locked = {false}
            closed = {true}
            />).toJSON()

        expect(tree).toMatchSnapshot()
    });

    test('should match snapshot', () => {
        const tree = renderer.create(<Display
            locked = {true}
            closed = {true}
            />).toJSON()

        expect(tree).toMatchSnapshot()
    });

    test('displays if gate is open/closed and if it is locked/unlocked', () => {
        const {getByText} = render(<Display
            locked={false}
            closed={false}
            />)
        
        getByText(/Unlocked/i)
        getByText(/Open/i)
    });

    test('displays `Closed` if the `closed` prop is `true` and `Open` if otherwise', () => {
        const {getByText, rerender} = render(<Display
            closed={true}
            />)
        getByText(/Closed/i)

        rerender(<Display
            closed={false}
            />)
        getByText(/Open/i)
    });

    test('displays `Locked` if the `locked` prop is `true` and `Unlocked` if otherwise', () => {
        const {getByText, rerender} = render(<Display
            locked={true}
            />)
        getByText(/Locked/i)

        rerender(<Display
            locked={false}
            />)
        getByText(/Unlocked/i)
    });
})