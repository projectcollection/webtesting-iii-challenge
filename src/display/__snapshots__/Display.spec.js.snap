// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display/> should match snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;

exports[`<Display/> should match snapshot 2`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;

exports[`<Display/> should match snapshot 3`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led red-led"
  >
    Closed
  </div>
</div>
`;

exports[`<Display/> should match snapshot 4`] = `
<div
  className="display panel"
>
  <div
    className="led red-led"
  >
    Locked
  </div>
  <div
    className="led red-led"
  >
    Closed
  </div>
</div>
`;
